test/Microsoft.Azure.ServiceBus.UnitTests/TokenProviderTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Azure.ServiceBus.UnitTests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.ServiceBus.Primitives;
+    using Microsoft.IdentityModel.Clients.ActiveDirectory;
+    using Xunit;
+
+    public class TokenProviderTests : SenderReceiverClientTestBase
+    {
+        /// <summary>
+        /// This test is for manual only purpose. Fill in the tenant-id, app-id and app-secret before running.
+        /// </summary>
+        /// <returns></returns>
+        [Fact]
+        [DisplayTestMethodName]
+        async Task UseITokenProviderWithAad()
+        {
+            var tenantId = "";
+            var aadAppId = "";
+            var aadAppSecret = "";
+
+            if (string.IsNullOrEmpty(tenantId))
+            {
+                TestUtility.Log($"Skipping test during scheduled runs.");
```

Comment:
Would consider adding more context - "Manual test requires authentication information. Skipping during automated runs.".

Reply:
On a second thought, that info is available as environment variables:
https://github.com/Azure/azure-service-bus-dotnet/blob/c58b25b4e242ae5744d690a345fb0b1d77428076/build/build.ps1#L19-L21

Reply:
We can automate the tests later. This requires creation of app identity and secrets as a ARM resource. We are a little in hurry to release the client tomorrow.

Reply:
@serkantkaraca that's exactly what I was implying. `appId` and `clientSecret` are created using ARM resource and stored in the environment for each build execution.